cmd/catalyst-node/catalyst-node.go

```diff
@@ -0,0 +1,284 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/golang/glog"
+	serfclient "github.com/hashicorp/serf/client"
+	"github.com/hashicorp/serf/cmd/serf/command/agent"
+	"github.com/mitchellh/cli"
+	"github.com/peterbourgon/ff/v3"
+)
+
+type (
+	catalystConfig struct {
+		serfRPCAddress           string
+		serfRPCAuthKey           string
+		mistLoadBalancerEndpoint string
+	}
+)
+
+var Commands map[string]cli.CommandFactory
+
+func init() {
+	ui := &cli.BasicUi{Writer: os.Stdout}
+
+	// eli note: this is copied from here:
+	// https://github.com/hashicorp/serf/blob/a2bba5676d6e37953715ea10e583843793a0c507/cmd/serf/commands.go#L20-L25
+	// but we should someday get a little bit smarter and invoke serf directly
+	// instead of wrapping their CLI helper
+
+	Commands = map[string]cli.CommandFactory{
+		"agent": func() (cli.Command, error) {
```

Comment:
What is the meaning of `agent` in the catalyst's nomenclature?

Reply:
The `agent` in the main serf process which handle the events and holds membership information.  
More here: https://www.serf.io/docs/agent/basics.html